pages/api/session.js

```diff
@@ -1,83 +1,65 @@
-      default:
-        throw new Error(`Unsupported method: ${req.method}`);
+    if (!email || !password) {
+        res.status(UNPROCESSABLE_ENTITY).json({ message: "Please enter your username/password." });
+        return
     }
-  } catch (error) {
-    await defaultErrorHandler(error, req, res);
-  }
-}
 
-async function login (req, res) {   
-  try {
+    const { session_token, user: { firstName, lastName, capabilities, defaultEntityType, forcePasswordChange } } = await People.login(email, password);
 
-    const {email=null, password=null} = req.body; 
-  
-    if (!email || !password) {
-      res.status(UNPROCESSABLE_ENTITY).json({message: "Please enter your username/password."});   
-      return
-      }
+    const token = generateToken({ firstName, lastName, email, capabilities, sessionToken: session_token });
+    const response = { message: 'Sucessfully logged in!' };
+
+    const { origin } = req.headers;
+    const isFromCMS = origin !== FRONTEND_URL;
 
-    const { session_token, user: {firstName, lastName, roles, capabilities, defaultEntityType, forcePasswordChange} } = await People.login(email, password);
-    req.session.session_token = session_token;
-    req.session.cache = {
-      firstName,
-      lastName,
-      roles,
-      capabilities,
-      defaultEntityType,
-      homepage: '/admin',
-      forcePasswordChange,
-    };
+    if (isFromCMS) {
+        const isAuthorized = capabilities.includes(canLogInToCms);
+        const errorMessage = "You do not have permission to log in. Please contact your administrator."
+        if (!isAuthorized) throw new UnauthorizedError(errorMessage);
 
-    const token = generateToken({ firstName, lastName });
-    setCookie('token', token, { req, res, httpOnly: true, secure: true, sameSite: 'none' });
+        req.session.session_token = session_token;
+        req.session.cache = {
+            token,
+            firstName,
+            lastName,
+            capabilities,
+            defaultEntityType,
+            homepage: '/admin',
+            forcePasswordChange,
+        };
 
-    await req.session.save();    
-    res.status(OK).json({ forcePasswordChange });
-  
-  } catch (error) {
-    if (error instanceof UnauthorizedError) {
-      res.status(UNPROCESSABLE_ENTITY).json({message: "Invalid username or password."});
-      return;
+        await req.session.save();
+
+        response.forcePasswordChange = forcePasswordChange;
     } else {
-      await defaultErrorHandler(error, req, res);
+        response.token = token;
     }
-  }
+
+    res.status(OK).json(response);
 }
 
-async function logout(req, res) {
-  try {
-    const session_token = assertUserIsLoggedIn(req);
-    await Session.logout(session_token); 
-    req.session.destroy();
-    res.status(200).json({message: 'OK'});
-  } catch (error) {
-    await defaultErrorHandler(error, req, res);
-  }
+async function del(req, res) {
+    const { origin } = req.headers;
```

Comment:
Kindly use assertUserCan at the beginning of all API endpoints.

https://github.com/klaudsol/klaudsol-commons/blob/main/src/lib/Permissions.js#L78